### PR TITLE
refactor: Rename ParseProviderNameWithVersion, move to relevant file, add test

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/908

This PR does a bit of refactoring around the code that parses the arguments passed to the `init` command, and add tests

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -38,12 +40,12 @@ var (
   cloudquery init aws gcp`,
 		Args: cobra.MinimumNArgs(1),
 		Run: handleCommand(func(ctx context.Context, _ *console.Client, cmd *cobra.Command, args []string) error {
-			return Initialize(ctx, args)
+			return initialize(ctx, args)
```
Comment (erezrokah) [inline cmd/init.go:43]: No need for this to be public

File: cmd/init.go [hunk 2]
```diff
@@ -56,7 +58,7 @@ func Initialize(ctx context.Context, providers []string) error {
 	rootBody := f.Body()
 	requiredProviders := make([]*config.RequiredProvider, len(providers))
 	for i, p := range providers {
-		organization, providerName, provVersion, err := registry.ParseProviderNameWithVersion(p)
+		organization, providerName, provVersion, err := parseProviderCLIArg(p)
```
Comment (erezrokah) [inline cmd/init.go:61]: I moved this function to this file and made it private. Also renamed to convey what it does

File: cmd/init.go [hunk 3]
```diff
@@ -154,6 +156,43 @@ func Initialize(ctx context.Context, providers []string) error {
 	return nil
 }
 
+// ParseProviderCLIArg parses the CLI arg passed to `cloudquery init <provider>`
+func parseProviderCLIArg(providerCLIArg string) (org string, name string, version string, err error) {
```
Comment (erezrokah) [inline cmd/init.go:160]: See original in https://github.com/cloudquery/cloudquery/pull/909/files#diff-3064fe03f4e76c66e781135a4e8b2c0165c5edb182bea64248daa68da95d6a10L37

I split the conditions in the original to make it easier to know the different formats and expected results

**Review by disq — APPROVED** [review]
Looks like a good idea

**Pull request merged**